x-pack/plugins/case/server/routes/api/cases/sub_case/patch_sub_cases.ts

```diff
@@ -0,0 +1,417 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import Boom from '@hapi/boom';
+import { pipe } from 'fp-ts/lib/pipeable';
+import { fold } from 'fp-ts/lib/Either';
+import { identity } from 'fp-ts/lib/function';
+import {
+  SavedObjectsClientContract,
+  KibanaRequest,
+  SavedObject,
+  SavedObjectsFindResponse,
+} from 'kibana/server';
+
+import { CaseClient } from '../../../../client';
+import { CASE_COMMENT_SAVED_OBJECT, SUB_CASE_SAVED_OBJECT } from '../../../../saved_object_types';
+import { CaseServiceSetup, CaseUserActionServiceSetup } from '../../../../services';
+import {
+  CaseStatuses,
+  SubCasesPatchRequest,
+  SubCasesPatchRequestRt,
+  CommentType,
+  excess,
+  throwErrors,
+  SubCasesResponse,
+  SubCasePatchRequest,
+  SubCaseAttributes,
+  ESCaseAttributes,
+  SubCaseResponse,
+  SubCasesResponseRt,
+  User,
+  CommentAttributes,
+} from '../../../../../common/api';
+import { SUB_CASES_PATCH_DEL_URL } from '../../../../../common/constants';
+import { RouteDeps } from '../../types';
+import {
+  AlertInfo,
+  escapeHatch,
+  flattenSubCaseSavedObject,
+  isGenOrAlertCommentAttributes,
+  wrapError,
+} from '../../utils';
+import { getCaseToUpdate } from '../helpers';
+import { buildSubCaseUserActions } from '../../../../services/user_actions/helpers';
+import { addAlertInfoToStatusMap } from '../../../../common';
+
+interface UpdateArgs {
+  client: SavedObjectsClientContract;
+  caseService: CaseServiceSetup;
+  userActionService: CaseUserActionServiceSetup;
+  request: KibanaRequest;
+  caseClient: CaseClient;
+  subCases: SubCasesPatchRequest;
+}
+
+function checkNonExistingOrConflict(
+  toUpdate: SubCasePatchRequest[],
+  fromStorage: Map<string, SavedObject<SubCaseAttributes>>
+) {
+  const nonExistingSubCases: SubCasePatchRequest[] = [];
+  const conflictedSubCases: SubCasePatchRequest[] = [];
+  for (const subCaseToUpdate of toUpdate) {
+    const bulkEntry = fromStorage.get(subCaseToUpdate.id);
+
+    if (bulkEntry && bulkEntry.error) {
+      nonExistingSubCases.push(subCaseToUpdate);
+    }
+
+    if (!bulkEntry || bulkEntry.version !== subCaseToUpdate.version) {
```

Comment:
Should we check of `bulkEntry.error` also?

Reply:
We check `bulkEntry.error` in the if block above right? I might be missing something do we need to check for it again?

Reply:
In the block above we check it but we do not `throw`. So on the second block you can still have `bulkEntry = {error: ...}`. The `!bulkEntry = false` so it will go the second part of the logical expression `bulkEntry.version !== subCaseToUpdate.version` but the object will not contain the `version` attribute.